gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAChallengeEndpoint.java

```diff
@@ -198,7 +198,10 @@ private void verifyCode(RoutingContext routingContext) {
             }
             // save enrolled factor if needed and redirect to the original url
             final MultiMap queryParams = RequestUtils.getCleanedQueryParams(routingContext.request());
-            final String returnURL = UriBuilderRequest.resolveProxyRequest(routingContext.request(), routingContext.get(CONTEXT_PATH) + "/oauth/authorize", queryParams, true);
+            final String returnURL =
```

Comment:
Maybe we can move this URL builder step to a common method into the AbstractEndpoint class in order to avoid missing implementation if we have to change something? wdyt?